__tests__/distributors/zulu-installer.test.ts

```diff
@@ -0,0 +1,369 @@
+        return {
+          statusCode: 200,
+          headers: {},
+          result: undefined
+        };
+      }
+    );
+    zuluDistributor = new ZuluDistributor({ version: '11', arch: 'x86', packageType: 'jdk' });
+    await expect(zuluDistributor['getAvailableVersions']()).rejects.toThrowError(
+      /No versions were found using url */
+    );
+  });
+});
+
+describe('getArchitectureOptions', () => {
+  it.each([
+    [{ architecture: 'x64' }, { arch: 'x86', hw_bitness: '64', abi: '' }],
+    [{ architecture: 'x86' }, { arch: 'x86', hw_bitness: '32', abi: '' }],
+    [{ architecture: 'x32' }, { arch: 'x32', hw_bitness: '', abi: '' }],
+    [{ architecture: 'arm' }, { arch: 'arm', hw_bitness: '', abi: '' }]
+  ])('%o -> %o', (input, expected) => {
+    let zuluDistributor = new ZuluDistributor({
+      version: '11',
+      arch: input.architecture,
+      packageType: 'jdk'
+    });
+    expect(zuluDistributor['getArchitectureOptions']()).toEqual(expected);
+  });
+});
+
+describe('findPackageForDownload', () => {
+  let spyHttpClient: jest.SpyInstance;
+  let zuluDistributor: ZuluDistributor;
+
+  beforeEach(() => {
+    spyHttpClient = jest.spyOn(HttpClient.prototype, 'getJson');
+    spyHttpClient.mockImplementation(
+      async (): Promise<ifm.ITypedResponse<any[]>> => {
+        manifestData = require('../data/zulu/zulu-releases-default.json');
+
+        const result = JSON.stringify(manifestData);
+        return {
+          statusCode: 200,
+          headers: {},
+          result: JSON.parse(result)
+        };
+      }
+    );
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+    jest.clearAllMocks();
+    jest.restoreAllMocks();
+  });
+
+  it.each([
+    // jdk
+
+    ['8.0.282', { version: '8', arch: 'x86', packageType: 'jdk' }],
+    ['11.0.10', { version: '11', arch: 'x86', packageType: 'jdk' }],
+    ['8.0.282', { version: '8.0', arch: 'x86', packageType: 'jdk' }],
+    ['11.0.10', { version: '11.0', arch: 'x86', packageType: 'jdk' }],
+    ['8.0.282', { version: '8', arch: 'x86', packageType: 'jdk+fx' }],
+    ['11.0.10', { version: '11', arch: 'x86', packageType: 'jdk+fx' }],
+    ['15.0.2', { version: '15', arch: 'x86', packageType: 'jdk' }],
+    ['15.0.2', { version: '15-ea', arch: 'x86', packageType: 'jdk' }]
+  ])('version is %s -> %o', async (expected, constructorInput) => {
+    let zuluDistributor = new ZuluDistributor(constructorInput);
+    const result = await zuluDistributor['findPackageForDownload'](zuluDistributor['version']);
+
+    expect(result.version).toBe(expected);
+  });
+
+  it('Should throw an error', async () => {
+    zuluDistributor = new ZuluDistributor({ version: '18', arch: 'x86', packageType: 'jdk' });
+    await expect(
+      zuluDistributor['findPackageForDownload'](zuluDistributor['version'])
+    ).rejects.toThrowError(/Could not find satisfied version for semver */);
+  });
+});
+
+describe('setupJava', () => {
```

Comment:
Let's get rid of setupJava tests since we have already tested them in base-installer

Reply:
Done. Thank you.